### ISSUE TITLE
Non-numeric value when clicking 'Cycle' tab

### ISSUE THREAD

**Issue by Axel577** [Cacti/plugin_cycle#10]
Hi!
Is anyone still contributing in the development of cycle?
I think it's a very useful plugin for admins who want to see all important graphs at one glance (with rotating sites)
I get this error message in cacti 1.2 beta 2:
![grafik](https://user-images.githubusercontent.com/28043745/47776722-5a99e980-dcf3-11e8-9458-207b9549ead8.png)
![grafik](https://user-images.githubusercontent.com/28043745/47776764-74d3c780-dcf3-11e8-95b8-fe3eb59d0b57.png)

**Comment by netniV** [comment]
We have not yet started to review all the plugins.  However, if this plugin worked for 1.0 or 1.1 it should also work for 1.2.

**Comment by Axel577** [comment]
mmh, doesn't work for me. It worked great in 1.1.38. I also reinstalled the plugin.
![grafik](https://user-images.githubusercontent.com/28043745/47786874-6eeae000-dd0d-11e8-8ee8-18ef9fb8242e.png)

But take your time, it's not that important.

**Comment by netniV** [comment]
There are some coding errors which I'm correcting now.  I'll publish but technically these should also fail in 1.1.38 too.

**Comment by netniV** [comment]
So I've already corrected the errors, but I'm seeing no graphs.  There may be an additional issue there.

**Comment by netniV** [comment]
OK, this should now be in the develop branch.  Feel free to download and try it.

**Comment by Axel577** [comment]
I've downloaded develop branch but I still get the old version:
![grafik](https://user-images.githubusercontent.com/28043745/47789664-70b8a180-dd15-11e8-9c7a-ebb1410d7cf1.png)

**Comment by netniV** [comment]
In fact, I've just found a major flaw in the javascript code.  It is reusing the name of an element as a variable.  I'm going to make some more changes.

**Comment by Axel577** [comment]
You got it!

**Comment by netniV** [comment]
Still getting no graphs though, so there's yet another issue ;-)

**Comment by netniV** [comment]
Actually, there isn't, that was purely me being a user ... I had no items in my tree 😱

**Comment by Axel577** [comment]
mmh, why do I get the same error message again :(
![grafik](https://user-images.githubusercontent.com/28043745/47791132-3e10a800-dd19-11e8-817c-b798813ffe93.png)

**Comment by Axel577** [comment]
There is no "plugins.php":
![grafik](https://user-images.githubusercontent.com/28043745/47791219-77491800-dd19-11e8-8d4e-24d54677c254.png)

**Comment by netniV** [comment]
That is occurring because there is some redirection occurring due to locks or errors... I noticed that `api_tree_lock` and `api_tree_unlock` both redirect to tree.php.  When I disabled that, it stopped the error.  Both redirections are wrong as they do not prefix the $config['url_path'] to the front to ensure that they are redirecting to the correct location.

That is a core issue and one I will be opening separately.  You may also find that the plugin has become disabled, hence you are seeing the plugins.php error.

**Comment by netniV** [comment]
So I don't seem to be getting any errors now with the develop version.  Try downloading again and see what you get.

**Comment by Axel577** [comment]
I've downloaded the lastest develop and get the same error:
![grafik](https://user-images.githubusercontent.com/28043745/47793533-c2b1f500-dd1e-11e8-911b-78f75122cbfe.png)

mmh..

**Comment by netniV** [comment]
Any errors in the log file?

**Comment by Axel577** [comment]
No, there are no errors in the cacti log. I've tried it with Firefox and IE. Doesn't make a difference.

**Comment by netniV** [comment]
Try clearing your browser cache then.  You see, I saw this issue with cycle.php because it's trying to call `applyFilter()`, the only problem is that something is redirecting the AJAX call to another location which then pulls down either tree.php or plugins.php with their own JavaScript code for `applyFilter()`.

That should only happen if you have an issue and it's redirecting you.  One way to find out would be to change the "header('Location: plugins.php" entries to "header('Location: ' . $config['url_path'] . 'plugins.php" (don't forget to add a `global $config;` inside the function that's doing the header call).

If that redirects you to the plugins page, it clear is moving you for some reason.

**Comment by netniV** [comment]
Just to point out, this isn't cycle's problem but caused by something else.  Cycle itself never redirect to plugins.php or tree.php

**Comment by Axel577** [comment]
Do you think that could be resolved with your cacti issue #2135?

**Comment by netniV** [comment]
Well, yes and no.  I'm still not clear how you are ending up at the plugins.php page, since the only redirections for that are enabling, disabling, installing, etc.  If you can, try temporarily disabling other plugins if you have any and see if the probably persists.

**Comment by Axel577** [comment]
I've disabled all plugins and cleared my browser cache. Sometimes, the error message appears, sometimes not. But I don't see any pictures. 
Furthermore: Next Update In ....  doesn't work. So I think there is a general problem. (display problem?)

![grafik](https://user-images.githubusercontent.com/28043745/47797202-2be93680-dd26-11e8-802d-d98f7f87aea5.png)

**Comment by netniV** [comment]
That would mean that your plugin isn't picking up the correct JS

**Comment by netniV** [comment]
Try doing this, go to settings -> Cycle, hit save.

**Comment by Axel577** [comment]
> 
> 
> Try doing this, go to settings -> Cycle, hit save.

Did it, no result.

Does this help:
![grafik](https://user-images.githubusercontent.com/28043745/47798536-deba9400-dd28-11e8-94a3-f6842f8fb020.png)

?

**Comment by netniV** [comment]
What browser are you using?

**Comment by Axel577** [comment]
omg, I solved it. But really by chance. When I first open the cycle tab, nothing happens. When I do a simple refresh of the page, it works 🙈
(Firefox 63.0.1)

**Comment by netniV** [comment]
So it still doesn't work until you refresh? Or now it's always working?

**Comment by Axel577** [comment]
It doesn't work until I do the refresh. But I have to do the refresh only once. When I switch to the "console" tab, for example, and back to "cycle" tab, it still works.
When I logout from cacti, and login again, it doesn't work. Then I have to do the refresh again. 
So it works after a refresh as long as I'm logged in.
But I'm okay with that. Now I know what I have to do ;-)
What do you think?

PS: But I think there is something else wrong with cycle.
My settings are those:
![grafik](https://user-images.githubusercontent.com/28043745/47842721-cfd1f100-ddbd-11e8-83ed-1d323394826c.png)

But they are never applied:
![grafik](https://user-images.githubusercontent.com/28043745/47842624-85507480-ddbd-11e8-8962-dffcd2e06809.png)

**Comment by netniV** [comment]
When you say 'Refresh' do you mean Browser Refresh or button 'Refresh'?

**Comment by Axel577** [comment]
> 
> 
> When you say 'Refresh' do you mean Browser Refresh or button 'Refresh'?

Browser Refresh

**Comment by netniV** [comment]
If it's browser, try clearing all cache's and then restarting firefox.  It should then pick up the script properly.

**Comment by Axel577** [comment]
I did that several times, but it didn't help. But I'm okay with that.
What do you think about the settings that don't get applied? Can you reproduce that?

**Comment by netniV** [comment]
Settings don't get applied?

**Comment by netniV** [comment]
```sql
 select * from settings where name like 'cycle_%';
+--------------------------+-------+
| name                     | value |
+--------------------------+-------+
| cycle_columns            | 2     |
| cycle_custom_graphs_list |       |
| cycle_custom_graphs_tree | 1     |
| cycle_custom_graphs_type | 2     |
| cycle_delay              | 60    |
| cycle_graphs             | 4     |
| cycle_height             | 100   |
| cycle_legend             |       |
| cycle_timespan           | 5     |
| cycle_width              | 400   |
+--------------------------+-------+
10 rows in set (0.00 sec)

select * from settings_user where name like 'cycle_%';
+---------+--------------------------+-------+
| user_id | name                     | value |
+---------+--------------------------+-------+
|       1 | cycle_columns            | 2     |
|       1 | cycle_custom_graphs_type | 2     |
|       1 | cycle_delay              | 60    |
|       1 | cycle_filter             |       |
|       1 | cycle_font_color         | 1     |
|       1 | cycle_font_face          |       |
|       1 | cycle_font_size          | 8     |
|       1 | cycle_graphs             | 4     |
|       1 | cycle_height             | 100   |
|       1 | cycle_legend             |       |
|       1 | cycle_timespan           | 5     |
|       1 | cycle_width              | 400   |
+---------+--------------------------+-------+
12 rows in set (0.00 sec)
```

**Comment by netniV** [comment]
This should now be resolved.  Please close if it is.

**Comment by Axel577** [comment]
I've downloaded the latest version and I get:
![grafik](https://user-images.githubusercontent.com/28043745/47846678-03664880-ddc9-11e8-9052-fb3cf05e1571.png)
![grafik](https://user-images.githubusercontent.com/28043745/47846732-34df1400-ddc9-11e8-853a-65962657e104.png)


Do the settings match in your setup?

**Comment by netniV** [comment]
Run the above SQL statements, i bet the plugin isn't handling user settings properly.

**Comment by netniV** [comment]
Check within the Edit Profile (top right) to see if cycle settings are in there too

**Comment by netniV** [comment]
Open that as a new issue if not.
